fe/free_energy_rabfe.py

```diff
@@ -32,23 +37,75 @@ def log(self):
             self.mol_name,
             "dG_complex_conversion (K complex)",
             self.dG_complex_conversion,
+            "dG_complex_conversion_err",
+            self.dG_complex_conversion_error,
             "dG_complex_decouple (E0 + A0 + A1 + E1)",
             self.dG_complex_decouple,
+            "dG_complex_decouple_err",
+            self.dG_complex_decouple_error,
             "dG_solvent_conversion (K solvent)",
             self.dG_solvent_conversion,
+            "dG_solvent_conversion_err",
+            self.dG_solvent_conversion_error,
             "dG_solvent_decouple (D)",
             self.dG_solvent_decouple,
+            "dG_solvent_decouple_err",
+            self.dG_solvent_decouple_error,
         )
 
     @classmethod
     def from_log(cls, log_line):
         """parse log line"""
         mol_name, rest = log_line.split("stage summary for mol: ")[-1].split(" dG_complex_conversion (K complex) ")
         tokens = rest.split()
-        value_strings = tokens[0], tokens[9], tokens[13], tokens[16]
-        values = list(map(float, value_strings))
+        values = [float(tokens[i]) for i in (0, 2, 11, 13, 17, 19, 22, 24)]
         return RABFEResult(mol_name, *values)
 
+    def __eq__(self, other: "RABFEResult") -> bool:
+        if not isinstance(other, RABFEResult):
+            return NotImplemented
+        equal = True
+        for field in fields(self):
+            self_val = getattr(self, field.name)
+            other_val = getattr(other, field.name)
+            # Python doesn't consider nan == nan to be true
+            if field.type is float and math.isnan(self_val) and math.isnan(other_val):
+                continue
+            equal &= self_val == other_val
+        return equal
```

Comment:
extreme nit: what do you think about a more functional style here, using `all`?
```suggestion
        def field_eq(field):
            self_val = getattr(self, field.name)
            other_val = getattr(other, field.name)
            if field.type is float and math.isnan(self_val) and math.isnan(other_val):
                return True
            return self_val == other_val

        return all(field_eq(field) for field in fields(self))
```